common/lib/util/multicaster.ts

```diff
@@ -0,0 +1,43 @@
+import Logger from './logger';
+
+class Multicaster {
+	members: Array<Function>;
+
+  // Private constructor; use static Multicaster.create instead
+	private constructor(members?: Array<Function>) {
+		this.members = members || [];
+	}
+
+	call(...args: unknown[]): void {
+		for (let i = 0; i < this.members.length; i++) {
+			let member = this.members[i];
+			if (member) {
+				try {
+          member(...args);
+				} catch (e) {
+					Logger.logAction(
+						Logger.LOG_ERROR,
+						'Multicaster multiple callback handler',
+						'Unexpected exception: ' + e + '; stack = ' + (e as Error).stack
+					);
+				}
+			}
+		}
+	}
+
+	push(...args: Array<Function>): void {
+		this.members.push(...args);
+	}
+
+	static create(members?: Array<Function>) {
+		const instance = new Multicaster(members);
+		return Object.assign(
+      (...args: unknown[]) => instance.call(args),
+      {
+        push: (fn: Function) => instance.push(fn)
```

Comment:
push can be updated as `push:instance.push` , same goes for `call` ig

Reply:
Nice spot, changed both in ed3209cb